File: backend/components/metiers/threed/tests_centralises/samples/validators/rgpd/test_rgpd_validators.js
// test_sample_rgpd.js
// Tests avancés pour sample_rgpd.js (validation RGPD)
const validateRGPD = require('../../../../samples/validators/rgpd/sample_rgpd');

describe('validateRGPD', () => {
  it('valide un objet avec hasConsent à true', () => {
    expect(validateRGPD({ hasConsent: true })).toBe(true);
  });
  it('rejette un objet avec hasConsent à false', () => {
    expect(validateRGPD({ hasConsent: false })).toBe(false);
  });
  it('rejette un objet sans hasConsent', () => {
    expect(validateRGPD({})).toBe(false);
  });
});

File: backend/components/metiers/threed/tests_centralises/samples/validators/rgpd/test_rgpd_validators.py
# test___init__.py
"""
Tests avancés pour sample_rgpd.py (validation RGPD).
"""
from threed.samples.validators.rgpd import sample_rgpd


class Dummy:
    has_consent = True


def test_validate_rgpd_true():
    d = Dummy()
    assert sample_rgpd.validate_rgpd(d) is True


def test_validate_rgpd_false():
    d = Dummy()
    d.has_consent = False
    assert sample_rgpd.validate_rgpd(d) is False


def test_validate_rgpd_missing():
    class NoConsent:
        pass

    assert sample_rgpd.validate_rgpd(NoConsent()) is False
